core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -997,9 +992,14 @@ public void sendResponse(Version nodeVersion, Channel channel, final TransportRe
                     transportServiceAdapter.onResponseSent(requestId, action, response, finalOptions);
                 }
             };
-            sendMessage(channel, reference, onRequestSent, false);
+            try {
+                sendMessage(channel, reference, onRequestSent, false);
```

Comment:
I wonder if we should have an internal private method that does this for use like:
`addedReleaseListener = sendMessageInternal(...)` so we do this only once?